lib/cocina/models.rb

```diff
@@ -148,9 +148,11 @@ def self.with_metadata(cocina_object, lock, created: nil, modified: nil)
     end
 
     def self.type_for(dyn)
-      dyn.with_indifferent_access.fetch('type')
-    rescue KeyError
```

Comment:
at least one full copy of the hash is made in memory here, which is not necessary

Reply:
Can you drop a comment in the code to this effect around here?

Reply:
i'd be happy to leave a code comment, though if this change is merged, the part of the code that makes the copy of the hash will be gone, so I'm not sure what the comment would say

Reply:
I think something like

```ruby
# Intentionally checking both string- and symbol-type keys in the hash via `#[]` instead of `#with_indifferent_access` (and/or `#fetch`) in order to be more memory-efficient
```

would be good, though I don't love my wording. Why? Lest we outsmart ourselves, or Rubocop tries the same, later on.

Reply:
I like that idea.  Added the comment. thanks